centreon/www/widgets/hostgroup-monitoring/src/index.php

```diff
@@ -97,161 +97,175 @@
 
 $serviceStateLabels = [0 => "Ok", 1 => "Warning", 2 => "Critical", 3 => "Unknown", 4 => "Pending"];
 
-$query = <<<'SQL_WRAP'
-    SELECT SQL_CALC_FOUND_ROWS DISTINCT
-        1 AS REALTIME, name, hostgroup_id
-    FROM hostgroups 
-    SQL_WRAP;
-
-if (isset($preferences['hg_name_search']) && $preferences['hg_name_search'] != "") {
-    $tab = explode(" ", $preferences['hg_name_search']);
-    $op = $tab[0];
-    if (isset($tab[1])) {
-        $search = $tab[1];
+try {
+    $query = <<<'SQL_WRAP'
+        SELECT SQL_CALC_FOUND_ROWS DISTINCT
+            1 AS REALTIME, name, hostgroup_id
+        FROM hostgroups 
+        SQL_WRAP;
+
+    if (isset($preferences['hg_name_search']) && $preferences['hg_name_search'] != "") {
+        $tab = explode(" ", $preferences['hg_name_search']);
+        $op = $tab[0];
+        if (isset($tab[1])) {
+            $search = $tab[1];
+        }
+        if ($op && isset($search) && $search != "") {
+            $query = CentreonUtils::conditionBuilder(
+                $query,
+                "name " . CentreonUtils::operandToMysqlFormat($op) . " '" . $dbb->escape($search) . "' "
+            );
+        }
     }
-    if ($op && isset($search) && $search != "") {
-        $query = CentreonUtils::conditionBuilder(
-            $query,
-            "name " . CentreonUtils::operandToMysqlFormat($op) . " '" . $dbb->escape($search) . "' "
-        );
+
+    if (!$centreon->user->admin) {
+        $query = CentreonUtils::conditionBuilder($query, "name IN (" . $aclObj->getHostGroupsString("NAME") . ")");
     }
-}
 
-if (!$centreon->user->admin) {
-    $query = CentreonUtils::conditionBuilder($query, "name IN (" . $aclObj->getHostGroupsString("NAME") . ")");
-}
+    $orderby = "name ASC";
+    if (isset($preferences['order_by']) && trim($preferences['order_by']) != "") {
+        $orderby = $preferences['order_by'];
+    }
 
-$orderby = "name ASC";
-if (isset($preferences['order_by']) && trim($preferences['order_by']) != "") {
-    $orderby = $preferences['order_by'];
-}
+    $query .= "ORDER BY :orderby";
+    $query .= " LIMIT " . ($page * $preferences['entries']) . "," . $preferences['entries'];
```

Comment:
Hi,
IMHO you also should bind params $preferences['entries']?
CodeRabbit seems to have made a consistent comment
@Drago37 & @nfragnet , what do you think about it?

Reply:
Yes, @selfakiri you need to do as orderby here using binding ;)

Reply:
good !